apps/backend/src/index.ts

```diff
@@ -6,27 +6,128 @@ import healthRoutes from "./routes/health";
+  IntersectionState, MapRequestToResponse, WebSocketErrorPayload,
+  WebSocketRequest,
+  WebSocketRequestTypes,
+  WebSocketResponse,
+  WebSocketResponseTypes
+} from "@types";
 const cors = require("cors");
+import { WebSocketServer, WebSocket } from 'ws';
+import { WebSocketRequestSchema } from '@types';
+import { Controller } from "@/lib/controller";
 
-const app = express();
+
+const httpServer = express();
 const corsOptions = {
   origin: `${FRONTEND_URL}`,
 };
-app.use(cors(corsOptions));
-app.use(express.json());
+httpServer.use(cors(corsOptions));
+httpServer.use(express.json());
 
 // Routes
-app.use("/api/user", userRoutes);
-app.use("/api/chip", chipRoutes);
-app.use("/api/oauth", oauthRoutes);
-app.use("/api/lanna", lannaRoutes);
-app.use("/api/message", messageRoutes);
-app.use("/", healthRoutes);
+httpServer.use("/api/user", userRoutes);
+httpServer.use("/api/chip", chipRoutes);
+httpServer.use("/api/oauth", oauthRoutes);
+httpServer.use("/api/lanna", lannaRoutes);
+httpServer.use("/api/message", messageRoutes);
+httpServer.use("/", healthRoutes);
 
-app.locals.intersectionState = {} as IntersectionState;
+httpServer.locals.intersectionState = {} as IntersectionState;
 
 const PORT = process.env.PORT || 8080;
-app.listen(PORT, () => {
+httpServer.listen(PORT, () => {
   console.log(`Server is running on port ${PORT}`);
 });
+
+// TODO: move into new file and hoist here
+const WS_PORT = process.env.WS_PORT || 8090;
+
+const wsServer = new WebSocketServer({ port: Number(WS_PORT) })
+
+const controller = new Controller();
+
+const clients: Record<string, WebSocket> = {};
+
+const handleError = (socket: WebSocket, target: string, error: string): void => {
+  const payload: WebSocketErrorPayload = {
+    error,
+  }
+
+  const resp: WebSocketResponse = {
+    type: WebSocketResponseTypes.ERROR,
+    targetSigPubKey: target,
+    senderSigPubKey: "",
+    payload
+  }
+
+  socket.send(JSON.stringify(resp))
+  return;
+}
+
+wsServer.on('connection', (socket: WebSocket) => {
+
+  socket.on('open', () => {
+    console.log("Open ws client connection")
+  });
+
+  socket.on('message', async (message: string) => {
+    const req: WebSocketRequest = WebSocketRequestSchema.parse(JSON.parse(message));
+
+    if (!req.authToken) {
+      return handleError(socket, req.senderSigPubKey, "Missing auth token.")
+    }
+
+    // Fetch user by auth token
+    // While the user isn't specifically required, it ensures the request is from an authenticated user
+    const user = await controller.GetUserByAuthToken(req.authToken);
+
+    if (!user) {
+      return handleError(socket, req.senderSigPubKey, "Invalid auth token.")
+    }
+
+    try {
+      switch (req.type) {
+        case WebSocketRequestTypes.CONNECT:
+          if (!req.senderSigPubKey) {
+            return handleError(socket, req.senderSigPubKey, "Missing target.")
+          }
+
+          // Set record for future lookup
+          clients[req.senderSigPubKey] = socket;
```

Comment:
Currently I use the signing public key as the identifier for a websocket. While this seemed like the simplest option, a more human-readable / friendly option would be to use username (which I believe is unique).